test/integration/deleted_services_and_plans_test.go

```diff
@@ -0,0 +1,115 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	// avoid error `servicecatalog/v1beta1 is not enabled`
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+
+	fakeosb "github.com/pmorie/go-open-service-broker-client/v2/fake"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/test/util"
+)
+
+func TestClusterServicePlanRemovedFromCatalogWithoutInstances(t *testing.T) {
+	_, catalogClient, _, _, _, _, shutdownServer, shutdownController := newTestController(t, fakeosb.FakeClientConfiguration{
+		CatalogReaction: &fakeosb.CatalogReaction{
+			Response: getTestCatalogResponse(),
+		},
+	})
+	defer shutdownController()
+	defer shutdownServer()
+
+	client := catalogClient.ServicecatalogV1beta1()
+
+	broker := &v1beta1.ClusterServiceBroker{
+		ObjectMeta: metav1.ObjectMeta{Name: testClusterServiceBrokerName},
+		Spec: v1beta1.ClusterServiceBrokerSpec{
+			URL: testBrokerURL,
+		},
+	}
+
+	_, err := client.ClusterServiceBrokers().Create(broker)
+	if nil != err {
+		t.Fatalf("error creating the broker %q (%q)", broker.Name, err)
+	}
+
+	err = util.WaitForBrokerCondition(client,
+		testClusterServiceBrokerName,
+		v1beta1.ServiceBrokerCondition{
+			Type:   v1beta1.ServiceBrokerConditionReady,
+			Status: v1beta1.ConditionTrue,
+		})
+	if err != nil {
+		t.Fatalf("error waiting for broker to become ready: %v", err)
+	}
+
+	err = util.WaitForClusterServiceClassToExist(client, testClusterServiceClassGUID)
+	if nil != err {
+		t.Fatalf("error waiting from ClusterServiceClass to exist: %v", err)
+	}
+
+	removedPlan := getTestClusterServicePlanRemoved()
+	removedPlan, err = client.ClusterServicePlans().Create(removedPlan)
+	if err != nil {
+		t.Fatalf("error creating ClusterServicePlan: %v", err)
+	}
+
+	err = util.WaitForClusterServicePlanToExist(client, testRemovedClusterServicePlanGUID)
+	if err != nil {
+		t.Fatalf("error waiting for ClusterServicePlan to exist: %v", err)
+	}
+
+	t.Log("updating ClusterServiceClass status")
+	removedPlan.Status.RemovedFromBrokerCatalog = true
+	_, err = client.ClusterServicePlans().UpdateStatus(removedPlan)
+	if err != nil {
+		t.Fatalf("error marking ClusterServicePlan as removed from catalog: %v", err)
+	}
+
+	err = util.WaitForClusterServicePlanToNotExist(client, testRemovedClusterServicePlanGUID)
+	if err != nil {
+		t.Fatalf("error waiting for remove ClusterServicePlan to not exist: %v", err)
+	}
+}
+
+const (
+	testRemovedClusterServicePlanGUID         = "removed-plan"
+	testRemovedClusterServicePlanExternalName = "removed-plan-name"
+)
+
+func getTestClusterServicePlanRemoved() *v1beta1.ClusterServicePlan {
```

Comment:
I think the naming here and with the constants is misleading; the plan returned from this function is just your happy ordinary everyday test plan, and we just happen to be setting it as removed later in a test. We also might want to use this in later integration tests, so could you name it `getTestClusterServicePlan`? And the constants `testClusterServicePlanX`.

Reply:
@pmorie agreed to do this in a follow-up.